toucan_connectors/one_drive/one_drive_connector.py

```diff
@@ -0,0 +1,116 @@
+import logging
+import os
+from pathlib import Path
+from typing import Optional
+
+import pandas as pd
+import requests
+from pydantic import Field
+
+from toucan_connectors.oauth2_connector.oauth2connector import (
+    OAuth2Connector,
+    OAuth2ConnectorConfig,
+)
+from toucan_connectors.toucan_connector import (
+    ConnectorSecretsForm,
+    ToucanConnector,
+    ToucanDataSource,
+)
+
+API_BASE_ROUTE: str = 'https://graph.microsoft.com/v1.0/me'
+AUTHORIZATION_URL: str = 'https://login.microsoftonline.com/common/oauth2/v2.0/authorize'
```

Comment:
This URL is only https://login.microsoftonline.com/common/oauth2/v2.0/authorize is available only if your microsoft account is a multitenant account

In case where the account has her own tenant, you need to change **common** by the tenant name

Reply:
This input could be a field in the connector